myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+        # Derivatives and their intermediary variables (i.e. dot(x) and dot_x)
+        # are handled in dudt_i, F_i and G_i; time is excluded from the output;
+        # pace is made Vc.
+        time = model.time()
+        pace = model.binding('pace')
+        special_vars = set(
+            [time]
+            + [v.rhs().var() for v in model.states()]
+            + [v.lhs().var() for v in model.states()]
+        )
+        if pace is not None:
+            special_vars.add(pace)
+
+        # Add metadata
+        export_lines.append('/*')
+        export_lines.append('This file was generated by Myokit.')
+        if model.meta:
+            export_lines.append('')
+            for key, val in sorted(model.meta.items()):
+                export_lines.append(f'{key}: {val}')
+            export_lines.append('')
+        export_lines.append('*/')
+        export_lines.append('')
+
+        # Add empty input parameter list
+        export_lines.append('/* Input parameters */')
+        export_lines.append('in = [ ]')
+        export_lines.append('')
+
+        # Add placeholder protocol
+        if pace is not None:
+            export_lines.append('/* Voltage protocol [mV] */')
+            export_lines.append(
+                'Vc { -80 + 120 * heaviside(t-500) - 80 * heaviside(t-1000) }'
+            )
+            export_lines.append('')
+
+        # Add constants
+        const_vars = (
+            set(model.variables(const=True, deep=True, state=False))
+            - special_vars
+        )
+        for component_label, eq_list in sorted_eqs.items():
+            const_eqs = [
+                eq for eq in eq_list.equations() if eq.lhs.var() in const_vars
+            ]
+            if const_eqs:
+                export_lines.append(f'/* Constants: {component_label} */')
+                for eq in const_eqs:
+                    v = eq.lhs.var()
+                    lhs = var_name(v)
+                    rhs = e.ex(eq.rhs)
+                    qname = v.qname()
+                    unit = '' if v.unit() is None else f' {v.unit()}'
+                    export_lines.append(
+                        f'{lhs} {{ {rhs} }} /* {qname}{unit} */'
+                    )
+                export_lines.append('')
+
+        # Add initial conditions `u_i`
+        export_lines.append('/* Initial conditions */')
+        export_lines.append('u_i {')
+        for v in model.states():
+            lhs = var_name(v)
+            rhs = myokit.float.str(v.initial_value(as_float=True))
+            qname = v.qname()
+            unit = '' if v.unit() is None else f' {v.unit()}'
+            export_lines.append(f'{tab}{lhs} = {rhs}, /* {qname}{unit} */')
+        export_lines.append('}')
+        export_lines.append('')
+
+        # Add initial conditions `dudt_i`
+        export_lines.append('dudt_i {')
+        for v in model.states():
+            lhs = var_name(v.rhs())  # Use intermediary dot_x instead of dot(x)
```

Comment:
Is this necessary for all eqs? I'd say it's quite rare for a `x = dot(y) + 1` to appear anywhere, so a bit annoying for users who don't need this functionality?

Reply:
It always writes `dot_x` to make name conversion more convenient.
`dot(x)` isn't written because that becomes `dot(x) = dot_x` during prep.
It will probably be the same output, but if that's rare I can stop adding intermediary variables for all states and handle derivative naming separately like in the EasyML converter.

Reply:
Only states with RHS references get intermediary variables in 1a212493

Reply:
Stopped creating intermediary variables in e5c3e65f (even for states with RHS references) as they seem to create runtime errors in DiffSL.

Reply:
Resumed creating intermediary variables in be18eabe as the real issue appears to be with DiffSL input ordering.